src/rpc/rawtransaction_util.cpp

```diff
@@ -149,18 +149,8 @@ static void TxInErrorToJSON(const CTxIn& txin, UniValue& vErrorsRet, const std::
     vErrorsRet.push_back(entry);
 }
 
-// TODO(https://github.com/bitcoin/bitcoin/pull/10973#discussion_r267084237):
-// The dependency on interfaces::Chain should be removed, so
-// signrawtransactionwithkey doesn't need access to a Chain instance.
-UniValue SignTransaction(interfaces::Chain& chain, CMutableTransaction& mtx, const UniValue& prevTxsUnival, CBasicKeyStore *keystore, bool is_temp_keystore, const UniValue& hashType)
+UniValue SignTransaction(CMutableTransaction& mtx, const UniValue& prevTxsUnival, CBasicKeyStore* keystore, std::map<COutPoint, Coin> coins, bool is_temp_keystore, const UniValue& hashType)
```

Comment:
Like @jnewbery said, could be `const std::map<COutPoint, Coin>& coins)`.

Reply:
Updated to pass-by-reference but I think can't be const due to ```coins[out] = std::move(newcoin)``` L198 ?

Reply:
I think you're right that it can't be `const`

Reply:
Then I think it should not pass by reference or you could comment that `coins` is modified after the call.

Reply:
@promag followed your suggestion and also took the opportunity to add comments for all SignTransaction params